source/slang/slang-ir-autodiff-unzip.h

```diff
@@ -277,14 +277,26 @@ struct DiffUnzipPass
 
     IRBlock* getUpdateBlock(IndexedRegion* region)
     {
-        // TODO: What if the 'continue' region has multiple
-        // blocks?
-        // We ideally want the _last_ block before control loops back.
+        auto initBlock = getInitializerBlock(region);
+
+        auto condBlock = region->firstBlock;
+
+        IRBlock* lastLoopBlock = nullptr;
+
+        for (auto predecessor : condBlock->getPredecessors())
+        {
+            if (predecessor != initBlock)
```

Comment:
Maybe assert when there are more than one predecessor other than the loop header?

Reply:
We should also add a call to `eliminateContinueBlocks` in `prepareForBackwardAutoDiff`.

Reply:
Yup, I'm prepping some tests for continue, I'll make this addition in that PR